### PR TITLE
Gh 2999/choose payment type

### PR DESCRIPTION
**Pull request by biafra23**
Handles #2999 and #2997

Changes proposed in this pull request:
- Passing relaylimit to sheet
- Add callbacks for user-selected-relayer
- Add callback for user-selected-signer

### CODE REVIEW

File: Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionContentViewController.swift [hunk 1]
```diff
@@ -41,6 +41,15 @@ class ReviewExecutionContentViewController: UITableViewController {
         self.transaction = transaction
     }
 
+    var userSelectedSigner: Bool {
+        set(value) {
```
Comment (MouazAlzahabi) [inline Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionContentViewController.swift]:
```suggestion
        set {
               builder.userSelectedSinger = newValue
        }
```

File: Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionViewController.swift [hunk 2]
```diff
@@ -131,6 +132,25 @@ class ReviewExecutionViewController: ContainerViewController, PasscodeProtecting
     @IBAction func didTapPaymentMethod(_ sender: Any) {
         // open payment method selection  
         let choosePaymentVC = ChoosePaymentViewController()
+        choosePaymentVC.remainingRelays = controller.remainingRelays
+        choosePaymentVC.chooseRelay = {
```
Comment (MouazAlzahabi) [inline Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionViewController.swift]:
```suggestion
        choosePaymentVC.chooseRelay = { [unowned self] in
```

It should be like this, then you can remove usage of self from the call back

File: Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionViewController.swift [hunk 3]
```diff
@@ -131,6 +132,25 @@ class ReviewExecutionViewController: ContainerViewController, PasscodeProtecting
...
+            LogService.shared.debug("User selected Relay")
+            self.userSelectedSigner = false
+
+            self.contentVC.userSelectedSigner = false
+            self.contentVC.reloadData()
+        }
+        choosePaymentVC.chooseSigner = {
```
Comment (MouazAlzahabi) [inline Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionViewController.swift]: Same here

File: Multisig/UI/Transaction/PaymentMethod/ChoosePaymentViewController.swift [hunk 4]
```diff
@@ -75,7 +96,17 @@ class ChoosePaymentViewController: UIViewController, UITableViewDelegate, UITabl
     private func buildRelayerCell(tableView: UITableView) -> UITableViewCell {
         let cell = tableView.dequeueCell(PaymentMethodCell.self)
         cell.accessoryType = .none
-        cell.setRelaying(4, ReviewExecutionViewController.MAX_RELAY_TXS)
+        cell.setRelaying(remainingRelays, ReviewExecutionViewController.MAX_RELAY_TXS)
         return cell
     }
 }
+
+extension UITableViewCell {
```
Comment (MouazAlzahabi) [inline Multisig/UI/Transaction/PaymentMethod/ChoosePaymentViewController.swift]: If the intention of this to make the cell not enabled then you can use: 
cell.isEnabled = false 
If the intention is to make it visually disabled then I think just setting subView.isEnabled = false will be enough no ?

File: Multisig/UI/Transaction/PaymentMethod/ChoosePaymentViewController.swift [hunk 5]
```diff
@@ -47,22 +50,40 @@ class ChoosePaymentViewController: UIViewController, UITableViewDelegate, UITabl
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        //TODO: pass relaying data; check payment option availability
         let cell = tableView.dequeueCell(BorderedInnerTableCell.self, for: indexPath)
+        //cell.selectionStyle = .gray
```
Comment (MouazAlzahabi) [inline Multisig/UI/Transaction/PaymentMethod/ChoosePaymentViewController.swift]: Please remove this commented code

**Review by MouazAlzahabi — APPROVED** [review]

File: Multisig/UI/Transaction/PaymentMethod/ChoosePaymentViewController.swift (hunk 4 above)
Comment (biafra23) [reply Multisig/UI/Transaction/PaymentMethod/ChoosePaymentViewController.swift]: IIRC tried ` cell.isEnabled = false` but somehow it didn't work. Will try once more

Comment (biafra23) [reply Multisig/UI/Transaction/PaymentMethod/ChoosePaymentViewController.swift]: This was my last unsuccessful attempt to make the cell visually disabled.

Comment (biafra23) [reply Multisig/UI/Transaction/PaymentMethod/ChoosePaymentViewController.swift]: Unfortunately there is no property isEnabled on that cell (BorderedInnerTableCell).

**Pull request merged**